app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java

```diff
@@ -65,4 +71,22 @@ public void onNewCategoryAdded(String categoryName, String categoryDescription)
     public void showAddCategoryScreen() {
         view.navigateToAddCategoryPage(contest);
     }
+
+    private void submitContest() {
+        view.showMessage(R.string.submitting_contest);
+        Disposable submitCall = restApi.submitContest(new ContestWrapper(contest.build()))
+                .compose(subscribeOnIoObserveOnUi())
+                .subscribe((this::onApiResult), throwable -> {
+                    Timber.d("API error creating contest " + throwable.getMessage());
+                    view.showMessage(R.string.error_api);
+
+                    //todo - remove when api endpoints are complete
+                    onApiResult(new ContestWrapper(contest.build()));
+                });
+        disposables.add(submitCall);
+    }
+
+    private void onApiResult(ContestWrapper response) {
```

Comment:
not sure if you really need this method since you're only using it in one place right now, but if you're planning for using it later elsewhere, it is fine.